src/app/static/src/app/store/modelOptions/mutations.ts

```diff
@@ -81,6 +78,56 @@ export const mutations: MutationTree<ModelOptionsState> = {
     },
 
     [ModelOptionsMutation.SetModelOptionsVersion](state: ModelOptionsState, action: PayloadWithType<VersionInfo>) {
-        state.version = action.payload
+        state.version = action.payload;
     }
 };
+
+function checkOptionsValid(formMeta: DynamicFormMeta): boolean {
+    let valid = true;
+    formMeta.controlSections.forEach(section => {
+        section.controlGroups.forEach(group => {
+            group.controls.forEach(control => {
+                // We could go further and check that if it is null or empty that this isn't
+                // a required control but just assuming this for now seems ok
+                if (control.value != null && control.value != "" && hasOptions(control)) {
+                    valid = valid && checkControlOptionValid(control);
+                }
+            })
+        })
+    })
+    return valid;
+}
+
+type ControlWithOptions = SelectControl | MultiSelectControl;
+
+function hasOptions(control: Control): control is ControlWithOptions {
+    return control.type === "select" || control.type === "multiselect";
+}
+
+function checkControlOptionValid(control: ControlWithOptions): boolean {
+    let valid = true;
+
+    const options = getAllOptions(control)
+    const value = control.value;
+    // Check string and array types, otherwise we assume valid
+    if (typeof value === 'string') {
+        valid = options.includes(value);
+    } else if (Array.isArray(value)) {
+        valid = value.every(item => options.includes(item));
+    }
+
+    return valid;
+}
+
+function getAllOptions(control: ControlWithOptions): string[] {
+    const options= control.options.map((option: Option) => getOptions(option));
+    return options.flat();
+}
+
+function getOptions(option: Option): string[] {
+    const options = [option.id];
+    if (option.children !== undefined) {
```

Comment:
this is for the nested area labels right? i think there were a decent amount with children as [] so makes sense to cut off those possibilities!
```suggestion
    if (option.children !== undefined && option.children.length > 0) {
```